specification.md

```diff
@@ -1075,9 +1077,167 @@ Note that the referenced function definition type in this case must be `rpc`.
 
 For more information about functions, reference the [Functions definitions](#Function-Definition) section.
 
+### Using Functions For GraphQL Service Invocations
+
+If you want to use GraphQL services, you can also invoke them using a similar syntax to the above methods with the caveat that your `functionRef`'s `arguments` property must have the following properties:
+
+| Parameter | Description | Type | Required |
+| --- | --- | --- | --- |
+| inputs | object whose keys are the field's arguments name, and its value that argument value | object | no |
+| selectionSet | A string containing a valid GraphQL [selection set](https://spec.graphql.org/June2018/#sec-Selection-Sets) | string | yes |
+
+And the outer `data` container object must be omitted from the response of successful queries and mutations, leaving only the selection set's specified fields. For exceptions the response must be returned in its entirety.
+
+We'll use the following [schema definition](https://graphql.org/learn/schema/) to show how that would work with both a query and a mutation:
+
+```graphql
+type Query {
+    pets: [Pet]
+    pet(id: Int!): Pet
+}
+
+type Mutation {
+    createPet(pet: PetInput!): Pet
+}
+
+type Treat {
+    id: Int!
+}
+
+type Pet {
+    id: Int!
+    name: String!
+    favoriteTreat: Treat
+}
+
+input PetInput {
+    id: Int!
+    name: String!
+    favoriteTreatId: Int
+}
+```
+
+#### Executing a `Query`
+
+In our workflow definition, we can then use a function definition for the `pet` query field as such:
+
+```json
+{
+  "functions": [
+    {
+      "name": "getOnePet",
+      "operation": "https://example.com/pets/graphql#query#pet",
+      "type": "graphql"
+    }
+  ]
+}
+```
+
+Note that the `operation` property has the following format for the `graphql` type:
+
+```text
+<url_to_graphql_endpoint>#<literal "mutation" or "query">#<mutation_or_query_field>
+```
+
+In order to invoke this query, we would use the following `functionRef` parameters:
+
+```json
+{
+  "refName": "getOnePet",
+  "arguments": {
+    "inputs": {
+      "id": 42
+    },
+    "selectionSet": "{ id, name, favoriteTreat { id } }"
+  }
+}
+```
+
+Which would return the following action state:
```

Comment:
Could we add "GraphQL" in this sentence so that its not confused with SW "actions" and SW workflow "state"? 
Maybe just use "following results"?

Reply:
Oh I meant result indeed!